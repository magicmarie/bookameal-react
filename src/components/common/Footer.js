import React from "react";
import moment from "moment";

const Footer = () => {
  // moment: js lib to manage dates
  // sets the year to the current year: 4-digits Year
  const year = moment().format("YYYY");
  return (
    <div className="footer">
      <p>
        Bookameal@
        {year}
      </p>
    </div>
  );
};
export default Footer;
