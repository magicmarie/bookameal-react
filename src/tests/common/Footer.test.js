import React from "react";
import { shallow } from "enzyme";
import Footer from "../../components/common/Footer";

describe("Footer component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Footer />);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
