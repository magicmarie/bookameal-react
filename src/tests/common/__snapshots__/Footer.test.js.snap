// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Bookameal@
        2018
</p>,
      "className": "footer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Bookameal@",
          "2018",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Bookameal@",
        "2018",
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Bookameal@
          2018
</p>,
        "className": "footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Bookameal@",
            "2018",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Bookameal@",
          "2018",
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
